pandas/core/index.py

```diff
@@ -111,9 +112,31 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=False,
         if fastpath:
             return cls._simple_new(data, name)
 
+        # RangeIndex pass-through
+        # Index(start, stop, ...) --> RangeIndex(start, stop, ...)
+        if isinstance(data, int):
+            if dtype is None and copy == False:
+                copy = None
+                range_constructor = True
+            elif isinstance(dtype, int):
+                range_constructor = True
+                if copy == False:
+                    copy = None
+                elif isinstance(copy, int):
+                    range_constructor = True
+                else:
```

Comment:
I would rather we only implement the range like construction when using RangeIndex explicitly. To make a range index from the index constructor, you could write something like `Index(range(...))` (or xrange on Python 2). This could call an alternate constructor, which might just be `_simple_new`. I'll follow up with a more explicit example when I'm not on my phone.